### DISCUSSION TITLE
Use open-webui in a different setup with Cloudflare pages and a proxy

### DISCUSSION THREAD

**Discussion by RonB** [open-webui/open-webui#1525] [Q&A]
Hello everyone,

I really like this project and would like to use it as a basis for specific customers that do not have technical knowledge of models etc, but can interact with tailormade services. A lot of the ui and components are perfect but for most users functionality like importing and adjusting modelfiles should be hidden based on their role.

My goal is to create a new or forked project that uses the main repo of open-webui but has different demands infrastructure wise. The setup is:
- client as a cloudflare pages application with sveltekit and the sveltekit cloudflare adapter. The app should proxy all requests to a the backend where the python backend server runs. I can use a cloudflare tunnel to that backend as it also serves as a public api. Authentication with openid using next-auth. Depending on the user authenticated I would also be able to change the ui and integrate functionality specific to the user,
- the backend serves not only as the the interface to ollama but also as the backend for all database requests from the client. The sveltekit project should proxy the db requests with the authorization token of the authorized user.

My main question is how to start this proces but not miss out on updates of the main repo. Should I fork the main repo and make sure my new code like new routes, adjustments in the package.json etc. do not interfere with the main repo so that I can stay up to date with the main repo and also be able to contribute with addition or updates to main branch via pull requests?

I would like to develop in a docker (vscode remote) development container so a Dockerfile for development should exist with specific needs for the development process.

Any suggestions on how to go about this would be greatly appreciated.

Thank you so much,
Ronald

**Comment by justinh-rahb** [comment]
Thank you @RonB for sharing your thoughts on the possibilities that could eventually be reality for Open WebUI! 

I'd like to propose an alternative to forking that might benefit both your goals and the broader community. We are enthusiastic about collaborative growth and believe that working within the existing project structure could avoid unnecessary fragmentation. The open-webui project already has ambitions and developmental pathways that align closely with what you're aiming to create. By contributing directly to the main project, your enhancements can become part of a larger, unified effort, possibly providing these features to an even wider audience.

Integrating your changes within the main repository can also simplify the process of staying synchronized with the most recent updates and developments without the complexities of managing a fork. This approach could offer a more sustainable long-term solution that benefits not only your specific implementation but enhances the core project as well.

We could work together to ensure the main project accommodates different user roles more seamlessly—including hiding or adapting certain functionalities as needed—which seems to align closely with your vision. Your idea to use role-based UI customizations is an excellent one, and I believe it would be a valuable addition to the project.

Moreover, working directly on the main repository ensures that any enhancement you develop can be maintained as part of the project’s ongoing evolution, shared with, and valued by, the entire community. Additionally, your contributions can help define key features and innovations that keep the project at the forefront of technology and user-centric design.
